src/outputs.js

```diff
@@ -1,10 +1,63 @@
+const { VALID_SUBNET_GROUPS } = require('./constants');
+
+function appendSubnets(subnetGroups, outputs) {
+  if (subnetGroups) {
+    const typesToNames = {
+      rds: 'RDSSubnetGroup',
+      redshift: 'ElastiCacheSubnetGroup',
+      elasticache: 'RedshiftSubnetGroup',
+      dax: 'DAXSubnetGroup',
+    };
+
+    const subnetOutputs = subnetGroups.map(subnetGroup => ({
+      [typesToNames[subnetGroup]]: {
+        Description: `Subnet Group for ${subnetGroup}`,
+        Value: [typesToNames[subnetGroup]],
+      },
+    }));
+
+    Object.assign(outputs, ...subnetOutputs);
+  }
+}
+
+function appendBastionHost(createBastionHost, outputs) {
+  if (createBastionHost) {
+    outputs.BastionSSHUser = {
+      Description: 'SSH username for the Bastion host',
+      Value: 'ec2-user',
+    };
+    outputs.BastionEIP = {
+      Description: 'Public IP of Bastion host',
+      Value: {
+        Ref: 'BastionEIP',
+      },
+    };
+  }
+}
+
+function appendExports(exportOutputs, outputs) {
+  if (exportOutputs) {
+    Object.entries(outputs).forEach(([name, value]) => {
+      // eslint-disable-next-line no-param-reassign
+      value.Export = { Name: { '!Join': ['-', ["!Ref 'AWS::StackName'", name]] } };
```

Comment:
Can you use ‘Fn::Join’ here instead of !Join? Same with removing !Ref. Otherwise looks good